virttest/libvirt_installer.py

```diff
@@ -0,0 +1,125 @@
+"""
+Installer code that implement KVM specific bits.
+
+See BaseInstaller class in base_installer.py for interface details.
+"""
+
+import os
+import platform
+from avocado.utils import process
+import base_installer
+
+
+__all__ = ['GitRepoInstaller', 'LocalSourceDirInstaller',
+           'LocalSourceTarInstaller', 'RemoteSourceTarInstaller']
+
+
+class LIBVIRTBaseInstaller(base_installer.BaseInstaller):
+
+    '''
+    Base class for libvirt installations
+    '''
+
+    def _set_install_prefix(self):
+        """
+        Prefix for installation of application built from source
+
+        When installing virtualization software from *source*, this is where
+        the resulting binaries will be installed. Usually this is the value
+        passed to the configure script, ie: ./configure --prefix=<value>
+        """
+        prefix = self.test_builddir
+        self.install_prefix = os.path.abspath(prefix)
+
+    def _install_phase_package(self):
+        """
+        Create libvirt package
+        """
+        if os.path.isdir("/root/rpmbuild/"):
+            process.system("rm -rf /root/rpmbuild/*")
+        process.system("make rpm")
+
+    def _install_phase_package_verify(self):
+        """
+        Check if rpms are generated
+        """
+        found = False
+        for fl in os.listdir('/root/rpmbuild/RPMS/%s/' % platform.machine()):
+            if fl.endswith('.rpm'):
+                found = True
+        if not found:
+            self.test.fail("Failed to build rpms")
+
+    def _install_phase_install(self):
+        """
+        Install libvirt package
+        """
+        process.system("rpm -Uvh --nodeps --replacepkgs --replacefiles --oldpackage /root/rpmbuild/RPMS/%s/libvirt*" % platform.machine())
+
+    def _install_phase_init(self):
+        """
+        Initializes the built and installed software
+
+        This uses a simple mechanism of looking up the installer name
+        for deciding what action to do.
+
+        :return: None
+        """
+        process.system("service libvirtd restart")
+
+    def _install_phase_init_verify(self):
+        """
+        Check if package install is success
```

Comment:
this is not been met by the method as the return value is not asserted to ensure it is success or failure.
This is common across all the methods

Reply:
sure,